config/routes.rb

```diff
@@ -34,6 +34,11 @@
       resources :u2f_authentications, only: %i(create)
       resources :totp_registrations, only: %i(new create destroy)
       resources :totp_authentications, only: %i(create)
+      resources :referral_promos, only: %i(create) do
```

Comment:
i think adding the `index` default action might do what `get :show` on collection is doing

`only: %i(create index)`